packages/graph/hash_graph/lib/graph/src/store/postgres/ontology/read.rs

```diff
@@ -46,3 +48,43 @@ where
             .change_context(QueryError)
     }
 }
+
+#[async_trait]
+impl<C: AsClient, T> crud::Read<'_, (), T> for PostgresStore<C>
+where
+    for<'de> T: OntologyDatabaseType + Deserialize<'de> + Send,
+{
+    type Output = Vec<T>;
+
+    async fn get(&self, _: ()) -> Result<Self::Output, QueryError> {
```

Comment:
I really am not sure how I feel about us using the unit-type as some magical indicator of "give me all". I know it's possibly somewhat consistent with not having a path segment in REST APIs but I really don't like it being inside our Postgres impl. It also just doesn't _seem_ particularly conventional.

Can't we encapsulate this in an enum, if we _**really**_ want to keep all of these impls in one over-generalised messy function

Reply:
As discussed off-GitHub there is no easy way to encapsulate it into an enumeration. In the long term we want to pass something like an AST, but this is not scoped enough to be implemented right now, so I introduced a unit-struct `AllLatest`, which acts as the identifier for now.

Reply:
(Also noting that we kind of hate this design but are going to change it soon as we tackle structural querying)